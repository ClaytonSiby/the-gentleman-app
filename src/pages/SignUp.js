import React from 'react';

const SignUp = () => (
  <div>Signing Up!</div>
);

export default SignUp;
