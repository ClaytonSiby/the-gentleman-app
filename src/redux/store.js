import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore } from 'redux-persist';
import persistReducer from './rootReducer';

const store = createStore(
  persistReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

const persistor = persistStore(store);

export { store, persistor };
