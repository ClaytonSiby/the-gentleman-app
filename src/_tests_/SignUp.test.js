import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import SignUp from '../pages/SignUp';
import store from './Suits.test';

describe('SignUp', () => {
  it('Renders the SignUp page successfully', () => {
    render(
      <Provider store={store}>
        <SignUp />
      </Provider>,
    );
  });

  test('renders consistently to the DOM', () => {
    const test = renderer.create(
      <Provider store={store}>
        <SignUp />
      </Provider>
    ).toJSON();

    expect(test).toMatchSnapshot();
  })
});
