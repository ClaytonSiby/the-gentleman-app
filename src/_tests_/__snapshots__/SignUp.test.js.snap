// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders consistently to the DOM 1`] = `
<div
  className="signup"
>
  <h1>
    Sign Up
  </h1>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="username"
      >
        Name
      </label>
      <input
        className="form-control"
        id="username"
        placeholder="Enter email"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="email"
        placeholder="Enter email"
        type="email"
      />
      <small
        className="text-muted form-text"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        placeholder="*********"
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="passwordConfirmation"
      >
        Password Confirmation
      </label>
      <input
        className="form-control"
        id="passwordConfirmation"
        placeholder="*********"
        type="password"
      />
    </div>
    <button
      className="my-3 bg-light border btn btn-default btn-block btn-md"
      disabled={false}
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
